python/taichi/lang/_ndarray.py

```diff
@@ -265,15 +265,16 @@ def __repr__(self):
 
 class NdarrayHostAccessor:
     def __init__(self, ndarray):
-        if _ti_core.is_real(ndarray.dtype):
+        dtype = ndarray.element_data_type()
+        if _ti_core.is_real(dtype):
```

Comment:
Nice to have: maybe let's put `is_real` in `utils.py` under `types`? It is also exported by pybind.

Reply:
done, thx!